src/config.rs

```diff
@@ -39,7 +41,24 @@ pub struct ConditionalConfig {
 }
 
 #[derive(Deserialize, Debug, PartialEq, Clone)]
-pub struct TemplateSlotsTable(pub HashMap<String, toml::Value>);
+pub struct TemplateSlotsTable(pub HashMap<String, TemplateSlots>);
```

Comment:
I wonder if it would not be much simpler to keep this `toml::Value` instead of `TemplateSlots` here, in order to later on go through that `HashMap`
and call for every value a `TryFrom::try_from` trait method (that needs to be implemented )

to give a brief illustration how I could imagine things would work out in a simpler way is this little abstraction in between:

```rust
/// a value coming from the config file
struct UserInputToml(toml::Value);
/// a value entered by a user on stdin
struct UserInputStdin(String);

/// keep a value from stdin and a template var definition
struct ValidateConvertInputStdin(UserInputStdin, TemplateSlots);
/// keep a toml value and a template var definition
struct ValidateConvertInputToml(UserInputToml, TemplateSlots);

/// final structure that has a value that has been validated
struct ValidatedValue(liquid::Object);

impl TryFrom<ValidateConvertInput> for ValidatedValue {
    type Error = ConversionError;

    fn try_from(value: ValidateConvertInput) -> std::result::Result<Self, Self::Error> {
        // todo: - validate the input value
        //       - convert it to a liquid::Object

        todo!()
    }
}

impl TryFrom<ValidateConvertInputStdin> for ValidatedValue {
    type Error = ConversionError;

    fn try_from(value: ValidateConvertInputStdin) -> std::result::Result<Self, Self::Error> {
        // todo: - validate the input value
        //       - convert it to a liquid::Object

        todo!()
    }
}
```